department-of-veterans-affairs/vets-api
# [API-39420] Add find POAs caching service

## Summary

* Use cache aside pattern
* Configure TTL to 24 hours
* Add tests

## Related issue(s)

[API-39420](https://jira.devops.va.gov/browse/API-39420)

## Testing done


To manually test:

1. Connect to BGS;
2. In a Rails console, run:
   ```
   require 'bgs_service/redis/find_poas_service'
   ClaimsApi::FindPOAsService.new.response
   ```
3. Observe the "slow" call (3–5 seconds) to BGS followed by the response;
4. Run `ClaimsApi::FindPOAsService.new.response` again;
5. Observe the "fast" call (~1 second) to Redis followed by the response;
6. Inspect Redis and observe the stored key-value pair:
    ![redis](https://github.com/user-attachments/assets/66d39677-bd89-4e81-8288-9d363e43f047)


## Screenshots

N/A

## What areas of the site does it impact?

New caching service for Local BGS > Find POAs call.

## Acceptance criteria


## Requested Feedback

We could probably abstract the `FindPOAsResponse` class if we plan on caching other BGS calls in a similar way. It only exists because the `CacheAside` class requires the response to implement `cache?`. We could also write our own `CacheAside` class that doesn't require a `cache?` method implementation 🤷

Also, if anyone knows how to test Redis other than "well, we didn't call BGS and got the response, so it must've come from somewhere," I am open to advice.
